File: Ruby/3E.rb
# Input: The input cosists of a single string, consisting of characters and numbers, the string's length doesn't exceed 200 characters.
# Output: Return the input string, but with all characters replaces with their respective positions in the alphabet.

# Example: "The sunset sets at twelve o' clock." => 20 8 5 19 21 14 19 5 20 19 5 20 19 1 20 20 23 5 12 22 5 15 3 12 15 3 11

def alphabetPosition (input)
    output = ""
    alphabet = "abcdefghijklmnopqrstuvwxyz"
    input = input.downcase
    chars = input.split('')
    chars.each do |c|
        if alphabet.include? c
            output = output + (alphabet.index(c) + 1).to_s + " "
        end
    end

    return output
end

File: NodeJS/3E.js
/* 
    Input: The input consists of a single string, consisting of numbers.
    Output: Return a single digit
    
    Example: iqTest('2 4 7 8 10') => 3 // Third number is odd, while the rest of the numbers are even
    Example: iqTest('1 2 1 1') => 2 // Second number is even, while the rest of the numbers are odd
*/

const iqTest = (numbers) => {
    const odd = numbers.split(' ').filter(n => n % 2 === 1);
    const even = numbers.split(' ').filter(n => n % 2 === 0);
    return odd.length > even.length ? numbers.split(' ').indexOf(even[0]) + 1 : numbers.split(' ').indexOf(odd[0]) + 1;
}

console.log(iqTest('2 4 7 8 10'));
console.log(iqTest('1 2 1 1'));